cms/static/js/spec/views/move_xblock_spec.js

```diff
@@ -0,0 +1,277 @@
+                        display_name: category,
+                        children: []
+                    },
+                    xblocks;
+
+                xblocks = options[category];
+                if (!xblocks) {
+                    return outline;
+                }
+
+                outline.child_info = cInfo; // eslint-disable-line no-param-reassign
+                _.each(_.range(xblocks), function(xblockIndex) {
+                    cInfo.children.push(
+                        createChildXBlockInfo(category, options, xblockIndex)
+                    );
+                });
+                return outline;
+            };
+
+            createCourseOutline = function(options) {
+                var courseOutline = {
+                    category: 'course',
+                    display_name: 'Demo Course',
+                    id: 'COURSE_ID_101'
+                };
+
+                return createXBlockInfo('section', options, courseOutline);
+            };
+
+            renderViews = function(courseOutlineJson, ancestorInfo) {
+                moveXBlockBreadcrumbView = new MoveXBlockBreadcrumbView({});
+                moveXBlockListView = new MoveXBlockListView(
+                    {
+                        model: new XBlockInfoModel(courseOutlineJson, {parse: true}),
+                        ancestorInfo: ancestorInfo || {ancestors: []}
+                    }
+                );
+            };
+
+            getDisplayedInfo = function() {
+                var viewEl = moveXBlockListView.$el;
+                return {
+                    categoryText: viewEl.find('.category-text').text().trim(),
+                    currentLocationText: viewEl.find('.current-location').text().trim(),
+                    xblockCount: viewEl.find('.xblock-item').length,
+                    xblockDisplayNames: viewEl.find('.xblock-item .xblock-displayname').map(
+                        function() { return $(this).text().trim(); }
+                    ).get(),
+                    forwardButtonSRTexts: viewEl.find('.xblock-item .forward-sr-text').map(
+                        function() { return $(this).text().trim(); }
+                    ).get(),
+                    forwardButtonCount: viewEl.find('.fa-arrow-right.forward-sr-icon').length
+                };
+            };
+
+            verifyListViewInfo = function(category, expectedXBlocksCount, hasCurrentLocation) {
+                var displayedInfo = getDisplayedInfo();
+                expect(displayedInfo.categoryText).toEqual(moveXBlockListView.categoriesText[category] + ':');
+                expect(displayedInfo.xblockCount).toEqual(expectedXBlocksCount);
+                expect(displayedInfo.xblockDisplayNames).toEqual(
+                    _.map(_.range(expectedXBlocksCount), function(xblockIndex) {
+                        return category + '_display_name_' + xblockIndex;
+                    })
+                );
+                if (category !== 'component') {
```

Comment:
I would extract this logic into separate method to verify information about currentLocation